tests/test_grammar_formal.py

```diff
@@ -0,0 +1,169 @@
+        # 'Expecting rule or terminal definition (missing colon)', 'a A\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..TOKEN., .A..', l.parse, 'a A\n')
+        # 'Illegal name for rules or terminals', 'Aa:\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..RULE., .a..', l.parse, 'Aa:\n')
+        # 'Alias expects lowercase name', 'a: -> "a"\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..STRING., ."a"..', l.parse, 'a: -> "a"\n')
+        # 'Unexpected colon', 'a::\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..COLON.,', l.parse, 'a::\n')
+        # 'Unexpected colon', 'a: b:\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..COLON.,', l.parse, 'a: b:\n')
+        # 'Unexpected colon', 'a: B:\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..COLON.,', l.parse, 'a: B:\n')
+        # 'Unexpected colon', 'a: "a":\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..COLON.,', l.parse, 'a: "a":\n')
+        # 'Misplaced operator', 'a: b??'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..OP., .\?..', l.parse, 'a: b??')
+        # 'Misplaced operator', 'a: b(?)'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..OP., .\?..', l.parse, 'a: b(?)')
+        # 'Misplaced operator', 'a:+\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..OP., .\+..', l.parse, 'a:+\n')
+        # 'Misplaced operator', 'a:?\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..OP., .\?..', l.parse, 'a:?\n')
+        # 'Misplaced operator', 'a:*\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..OP., .\*..', l.parse, 'a:*\n')
+        # 'Misplaced operator', 'a:|*\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..OP., .\*..', l.parse, 'a:|*\n')
+        # 'Expecting option ("|") or a new rule or terminal definition', 'a:a\n()\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..LPAR.,', l.parse, 'a:a\n()\n')
+        # 'Terminal names cannot contain dots', 'A.B\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..TOKEN., .B..', l.parse, 'A.B\n')
+        # 'Expecting rule or terminal definition', '"a"\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..STRING., ."a"..', l.parse, '"a"\n')
+        # '%import expects a name', '%import "a"\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..STRING., ."a"..', l.parse, '%import "a"\n')
+        # '%ignore expects a value', '%ignore %import\n'
+        self.assertRaisesRegex(UnexpectedToken, 'Unexpected token Token..__ANON_2., .%import..', l.parse, '%ignore %import\n')
+
+    # def test_empty_literal(self):
```

Comment:
Why all the commented out parts?

Reply:
I wanted it to be clear which `test_grammar` tests weren't implementable for `lark.lark`.  Sort of a "_show your work_" thing.  But I can remove them.

Reply:
Fixed in 9493f81e9eab6dbe42096938bbcfc1f0cf5a2135.

Reply:
I appreciate your aim for completeness, but I think it was more confusing than anything.

If you think it's important information, perhaps just write a list of the names, without Python syntax. like

```python
# List of non-testable constructs: empty literal, kraken, foobar, etc.
```